File: data/imported/knowledge/slides/java/merged/snippets/snippet_004.java
@Test
    void testAddComment() throws IOException {
        try (var pres = new Presentation()) {
            ICommentAuthor author = pres.getCommentAuthors().addAuthor("Alice", "A");
            ISlide slide = pres.getSlides().get(0);
            LocalDateTime now = LocalDateTime.of(2026, 1, 15, 12, 0, 0);
            IComment comment = author.getComments().addComment("Review note", slide,
                    new PointF(2.0f, 3.0f), now);
            assertThat(comment.getText()).isEqualTo("Review note");
            assertThat(comment.getAuthor().getName()).isEqualTo("Alice");

            try (var pres2 = saveAndReopen(pres)) {
                ICommentAuthor a2 = pres2.getCommentAuthors().get(0);
                assertThat(a2.getComments().size()).isEqualTo(1);
                assertThat(a2.getComments().get(0).getText()).isEqualTo("Review note");
            }
        }
    }

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_004.cpp
TEST(CommentsIntegrationTestNoFixture, GetSlideComments) {
    Presentation pres;
    auto& a1 = pres.comment_authors().add_author("Alice", "A");
    auto& a2 = pres.comment_authors().add_author("Bob", "B");
    auto& slide = pres.slides()[0];
    auto now = std::chrono::system_clock::now();

    a1.comments().add_comment("Alice's", slide, PointF(1, 1), now);
    a2.comments().add_comment("Bob's", slide, PointF(2, 2), now);

    auto all_c = slide.get_slide_comments(nullptr);
    EXPECT_EQ(all_c.size(), 2);

    auto bob_c = slide.get_slide_comments(&a2);
    ASSERT_EQ(bob_c.size(), 1);
    EXPECT_EQ(bob_c[0]->text(), "Bob's");
}